port/osx/omrvmem.c

```diff
@@ -676,32 +676,43 @@ getMemoryInRange(struct OMRPortLibrary *portLibrary, struct J9PortVmemIdentifier
 			currentAddress = (void *)alignmentInBytes;
 		}
 	}
-
-	if (NULL == memoryPointer) {
-		/* Try all addresses within range */
-		while ((startAddress <= currentAddress) && (endAddress >= currentAddress)) {
-			memoryPointer = reserveMemory(portLibrary, currentAddress, byteAmount, identifier, mode, pageSize, pageFlags, category);
-
-			if (NULL != memoryPointer) {
-				/* stop if returned pointer is within range */
-				if ((startAddress <= memoryPointer) && (endAddress >= memoryPointer)) {
-					break;
-				}
-				if (0 != omrvmem_free_memory(portLibrary, memoryPointer, byteAmount, identifier)) {
-					return NULL;
-				}
-				memoryPointer = NULL;
+	
+#if defined(VM_FLAGS_SUPERPAGE_SIZE_2MB)
+	if (PPG_vmem_pageSize[2] == pageSize) {
+		/*
+		 * Don't bother scanning the entire address space if the request can't be satisfied.
+		 */
+		void *tmpPointer = reserveMemory(portLibrary, NULL, byteAmount, identifier, mode, pageSize, pageFlags, category);
```

Comment:
Note: This is the new code. The remaining is reformatted code. Passing `NULL (= address)` to `reserveMemory` triggers an `mmap` with `NULL (= address)`. Here, the system tries to automatically select the `address`. If `mmap` fails, then there is no memory available. Assuming `mmap`'s fail path is quick, this should improve perf when no memory is available.

Which benchmark was used to verify the perf improvement?

Reply:
On a Mac if you run `java -Xlp:objectheap:pagesize=2m` it takes a really really long time to fail to allocate 2m and start java.

Reply:
Not the Windows omvvmem.c does a similar check.